extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -286,10 +286,15 @@ export class AmpConsent extends AMP.BaseElement {
       let isConsentRequiredPromise;
 
       if (this.consentConfig_[instanceId]['checkConsentHref']) {
-        isConsentRequiredPromise = this.getConsentRemote_(instanceId).then(
-            response => {
-              this.parseConsentResponse_(instanceId, response);
-            });
+        const remoteConsentPromise = this.getConsentRemote_(instanceId);
+        isConsentRequiredPromise = remoteConsentPromise.then(response => {
+          return this.parseConsentResponse_(instanceId, response);
+        });
+        const sharedDataPromise = remoteConsentPromise.then(response => {
+          return response ? response['sharedDataPromise'] : null;
+        });
+        this.consentStateManager_.setConsentInstanceSharedData(
```

Comment:
why we pass the sharedDataPromise here, but not just sharedData?

Reply:
This is to completely avoid risks of race condition later on.

Reply:
Actually if we are going to support suppress consent state, we will need to wait for response from remote endpoint before resolve any consent policy. In that case there won't be any race condition. 
But if you want to add an extra layer of protection, this also looks good.

Reply:
Yep. I think this is a safer way to do it.